akka-actor/src/main/scala/akka/serialization/Serialization.scala

```diff
@@ -123,12 +124,18 @@ class Serialization(val system: ExtendedActorSystem) extends Extension {
           if (manifest == "")
             s1.fromBinary(bytes, None)
           else {
-            system.dynamicAccess.getClassFor[AnyRef](manifest) match {
-              case Success(classManifest) ⇒
-                s1.fromBinary(bytes, Some(classManifest))
-              case Failure(e) ⇒
-                throw new NotSerializableException(
-                  s"Cannot find manifest class [$manifest] for serializer with id [$serializerId].")
+            val cachedClassManifest = manifestCache.get(manifest)
```

Comment:
Add a FIXME that once compression is enabled it will also serve as a cache, i.e. this cache lookup will be simply an array indexing into the comression table that stores the actual instances. For the not-common manifests we can still do the slow lookup without caching then.